python/Converters/PycBasicData.h

```diff
@@ -99,19 +99,24 @@ namespace casacore { namespace python {
       PyObject* obj_ptr,
       boost::python::converter::rvalue_from_python_stage1_data* data)
     {
-#ifdef IS_PY3K
-    PyObject * temp_bytes = PyUnicode_AsEncodedString(obj_ptr, "ASCII", "strict"); // Owned reference
-    char* value;
-    if (temp_bytes != NULL) {
-        value = PyBytes_AS_STRING(temp_bytes); // Borrowed pointer
-        value = strdup(value);
-        Py_DECREF(temp_bytes);
+    char* value = 0;
+    if (PyUnicode_Check(obj_ptr)) {
+        PyObject * temp_bytes = PyUnicode_AsEncodedString(obj_ptr, "UTF-8", "strict"); // Owned reference
+
+        if (temp_bytes != NULL) {
+            value = PyBytes_AS_STRING(temp_bytes); // Borrowed pointer
+            value = strdup(value);
+            Py_DECREF(temp_bytes);
+        } else {
+          boost::python::throw_error_already_set();
```

Comment:
Mis-indented.

Reply:
yeah indentation was a mess, tried to fix it but it is still messy.

Reply:
i think the project needs a reformatting PR, going to leave this now.